src/commands/team.rs

```diff
@@ -1,163 +1,690 @@
+                    }
+                };
+
+                let data_analysis = if let Some(data_analysis) = &self.data_analysis {
+                    Some(Ok(data_analysis.clone()))
+                } else {
+                    if let Some(team) = &self.team {
+                        if team.program.code == Some("VRC".to_string()) {
+                            Some(vrc_data_analysis.team_info(&team.number).await)
+                        } else {
+                            None
+                        }
+                    } else {
+                        return CreateEmbed::new()
+                            .title("Invalid team data.");
+                    }
+                };
+
+                embed = embed
+                    .title(format!(
+                        "{} ({}, {}) Statistics",
+                        team.number, team.program.code.clone().unwrap_or("VRC".to_string()), team.grade
+                    ));
+
+                match skills_ranking {
+                    Ok(ranking) => {
+                        if let Some(ranking) = ranking {
+                            embed = embed.field(
+                                "Skills",
+                                format!(
+                                    "World Ranking: **#{}**\nDriver: **{}**\nProgramming: **{}**\nCombined: **{}**",
+                                    ranking.rank,
+                                    ranking.scores.max_driver,
+                                    ranking.scores.max_programming,
+                                    ranking.scores.score
+                                ),
+                                false
+                            );
+                        } else {
+                            embed = embed.field("Skills", "No skills runs found for this team.", false);
+                        }
+                    },
+                    Err(err) => {
+                        embed = embed.field("Failed to get skills ranking data from RobotEvents.", format!("```rs\n{err:?}```"), false);
+                    }
                 }
+
+                if let Some(analysis_result) = data_analysis {
+                    match analysis_result {
+                        Ok(analysis) => {
+                            embed = embed.field(
+                                "TrueSkill",
+                                    format!(
+                                        "TrueSkill: **{}**\nWorld Ranking **#{}**",
+                                        analysis.trueskill,
+                                        analysis.trueskill_ranking
+                                    ),
+                                    false
+                                )
+                                .field(
+                                    "Match Statistics",
+                                    format!(
+                                        "OPR: **{}**\nDPR: **{}**\nCCWM: **{}**",
```

Comment:
Yeah, it's not exactly ideal although i'm not sure how to structure this necessarily better. There are just a lot of things needing to be unwrapped since data analysis/skills data availability is dependent on a lot of conditional factors.